### PR TITLE
updated cobra and switched some code to use Args concept

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -49,10 +50,8 @@ var analyzeCmd = &cobra.Command{
 }
 
 func checkAnalyzeArgNum(args []string) error {
-	var errMessage string
 	if len(args) != 1 {
-		errMessage = "'analyze' requires one image as an argument: container analyze [image]"
-		glog.Errorf(errMessage)
+		errMessage := "'analyze' requires one image as an argument: container analyze [image]"
```
Comment (nkubala) [inline cmd/analyze.go]: nit: don't even need to make this var here, just instantiate the new error with the string directly

**Comment by aaron-prindle** [comment]
Incorporated feedback, this should be ready to merge

File: cmd/version.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ var versionCmd = &cobra.Command{
 	Use:   "version",
 	Short: "Print the version of container-diff",
 	Long:  `Print the version of container-diff.`,
+	Args:  cobra.ExactArgs(0),
```
Comment (r2d4) [inline cmd/version.go]: Nice, we should use this in minikube to replace some boilerplate

File: cmd/analyze.go [hunk 3]
```diff
@@ -32,15 +32,16 @@ var analyzeCmd = &cobra.Command{
 	Use:   "analyze",
 	Short: "Analyzes an image: [image]",
 	Long:  `Analyzes an image using the specifed analyzers as indicated via flags (see documentation for available ones).`,
-	Run: func(cmd *cobra.Command, args []string) {
+	Args: func(cmd *cobra.Command, args []string) error {
 		if err := validateArgs(args, checkAnalyzeArgNum, checkArgType); err != nil {
```
Comment (r2d4) [inline cmd/analyze.go]: Lets make an issue to replace the "arg count and type checking" to be their own flag types.

**Pull request merged**